lib/metrics.go

```diff
@@ -322,7 +503,58 @@ func (m *Metrics) UpdateBlockMetrics(proposerAddress []byte, txCount int, durati
 		m.ProposerCount.Inc()
 	}
 	// update the number of transactions
-	m.TransactionCount.Add(float64(txCount))
+	m.BlockNumTxs.Add(float64(txCount))
 	// update the block processing time in seconds
 	m.BlockProcessingTime.Observe(duration.Seconds())
+	// update block size
+	m.BlockSize.Set(float64(blockSize))
+	// update the block vdf iterations
+	m.BlockVDFIterations.Set(float64(vdfIterations))
+}
+
+// UpdateMempoolMetrics() updates mempool telemetry
+func (m *Metrics) UpdateMempoolMetrics(txCount, size int) {
+	// exit if empty
+	if m == nil {
```

Comment:
I didn't catch this on the last review but on what case m could be `nil`? In theory you could call the method by passing `m` as `nil` but I don't see a case where that would be done

Reply:
So if `Metrics` is set to nil, then these calls in code won't fail. (Like in testing or if disabled)

It's an alternative to nil checking everytime you do a call to metrics.